ppl/detail/flows/getCorrelationQuery.ts

```diff
@@ -0,0 +1,21 @@
+import {
+  cidCorrelation,
+  connCorrelation,
+  uidCorrelation
+} from "src/js/searches/programs"
+import {zng} from "zealot"
+import {Correlation} from "../models/Correlation"
+
+export function getCorrelationQuery(record: zng.Record) {
+  const {uid, cid} = new Correlation(record).getIds()
+
+  if (cid && uid) {
+    return connCorrelation(record)
+  } else if (uid) {
+    return uidCorrelation(uid)
+  } else if (cid) {
+    return cidCorrelation(cid)
+  } else {
+    return null
+  }
+}
```

Comment:
I needed to split this out so that, if there are more than 100 events returned, we can offer a link that says, "Query for all". In which case it calls this function to get the query and put it in the main search input.